source/header.c

```diff
@@ -354,6 +354,8 @@ int aws_cryptosdk_priv_hdr_parse_frame_len(
 
 int aws_cryptosdk_priv_hdr_parse_alg_suite_data(
     struct aws_cryptosdk_hdr *hdr, const struct aws_cryptosdk_alg_properties *alg_props, struct aws_byte_cursor *cur) {
+    AWS_PRECONDITION(aws_cryptosdk_hdr_is_valid(hdr));
```

Comment:
Don't you needs `alg_props` to be valid here? I see the assumption in the proof but not here.

Reply:
Added!